python/lsst/sims/photUtils/PhotometricParameters.py

```diff
@@ -2,50 +2,177 @@
+                 platescale=None,
+                 sigmaSys=None,
+                 bandpass=None):
 
         """
-        @param [in] exptime exposure time in seconds (default 15)
+        @param [in] exptime exposure time in seconds (defaults to LSST value)
+
+        @param [in] nexp number of expusres (defaults to LSST value)
+
+        @param [in] effarea effective area in cm^2 (defaults to LSST value)
+
+        @param [in] gain electrons per ADU (defaults to LSST value)
+
+        @param [in] readnoise electrons per pixel per exposure (defaults to LSST value)
 
-        @param [in] nexp number of expusres (default 2)
+        @param [in] darkcurrent electons per pixel per second (defaults to LSST value)
 
-        @param [in] effarea effective area in cm^2 (default for 6.5 meter diameter)
+        @param [in] othernoise electrons per pixel per exposure (defaults to LSST value)
 
-        @param [in] gain electrons per ADU (default 2.3)
+        @param [in] platescale arcseconds per pixel (defaults to LSST value)
 
-        @param [in] readnoise electrons per pixel per exposure (default 5.0)
+        @param [in] sigmaSys systematic error in magnitudes
+        (defaults to LSST value)
 
-        @param [in] darkcurrent electons per pixel per second (default 0.2)
+        @param [in] bandpass is the name of the bandpass to which these paramters
+        correspond.  If set to an LSST bandpass, the constructor will initialize
+        PhotometricParamters to LSST default values for that bandpass, excepting
+        any paramters that have been set by hand, i.e
 
-        @param [in] othernoise electrons per pixel per exposure (default 4.69)
+        myPhotParams = PhotometricParameters(nexp=3, bandpass='u')
 
-        @param [in] platescale arcseconds per pixel (default 0.2)
+        will initialize a PhotometricParameters object to u bandpass defaults, except
+        with 3 exposures instead of 2.
+
+        If bandpass is left as None, other parameters will default to LSST r band
+        values (except for those values set by hand).  The bandpass member variable
+        of PhotometricParameters will, however, remain None.
         """
 
-        self._exptime = exptime
-        self._nexp = nexp
-        self._effarea = effarea
-        self._gain = gain
-        self._platescale = platescale
+        # readnoise, darkcurrent and othernoise are measured in electrons.
+        # This is taken from the specifications document LSE-30 on Docushare
+        # Section 3.4.2.3 states that the total noise per pixel shall be 12.7 electrons
+        # which the defaults sum to (remember to multply darkcurrent by the number
+        # of seconds in an exposure=15).
+
+        self._exptime = None
+        self._nexp = None
+        self._effarea = None
+        self._gain = None
+        self._platescale = None
+        self._sigmaSys = None
+        self._readnoise = None
+        self._darkcurrent = None
+        self._othernoise = None
+
```

Comment:
I just started fixing this and realized that you and I have slightly different ideas for how the values should be checked.  What I wrote does this:

If the user specifies a value for a parameter by hand, use that.
If the user specified a bandpass that exists in DefaultPhotometricParameters, use that.
If the user did not specify a bandpass, use the 'any' key in DefaultPhotometricParameters.
If the user specified a bandpass that does not exist in DefaultPhotometricParameters, do nothing.

Thus, if the user specified a bandpass like 'y4', or 'sdss_u', then everything would be left as None (unless they specified the parameters by hand) and an exception would be raised.

I feel like you are suggesting:

If the user specified a value by hand, use that.
If the user specified a valid bandpass, use that.
If the user specified an invalid bandpass, use the 'any' key.
If the user did not specify a bandpass, do nothing.

I think I the first flow because people who don't want to think about it can just do

myPhotParams = PhotometricParameters()

and get the default values.  Does this sound reasonable?